extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java

```diff
@@ -600,7 +599,7 @@ private ValuesReader getDataReader(Encoding dataEncoding, ByteBuffer in, int val
         }
         ValuesReader dataReader;
         if (dataEncoding.usesDictionary()) {
-            final Dictionary dictionary = dictionarySupplier.get();
+            final Dictionary dictionary = dictionarySupplier.apply(context);
```

Comment:
For materializing the dictionary, we need a context.
That is what has complicated this change because dictionaries are materialized on demand only if needed. 
Please let me know if you have any suggestions to simplify this part.

Reply:
I'll think about it, but nothing better comes to mind yet.